src/control/cmd/daos/util.go

```diff
@@ -153,33 +152,36 @@ func createWriteStream(ctx context.Context, prefix string, printLn func(line str
 	}
 
 	go func(ctx context.Context, prefix string) {
-		if prefix != "" {
-			prefix = ": "
-		}
 
 		rdr := bufio.NewReader(r)
 		for {
-			select {
-			case <-ctx.Done():
-				return
-			default:
-				line, err := rdr.ReadString('\n')
-				if err != nil {
-					if !(errors.Is(err, io.EOF) || errors.Is(err, os.ErrClosed)) {
-						printLn(fmt.Sprintf("read err: %s", err))
-					}
-					return
+			line, err := rdr.ReadString('\n')
+			if err != nil {
+				if !(errors.Is(err, io.EOF) || errors.Is(err, os.ErrClosed)) {
+					printLn(fmt.Sprintf("read err: %s", err))
 				}
-				printLn(fmt.Sprintf("%s%s", prefix, line))
+				return
+			}
+			if line == "close\n" {
+				w.Close()
+				r.Close()
+				return
+			}
+			if prefix == "" {
+				printLn(fmt.Sprintf("%s", line))
+			} else {
+				printLn(fmt.Sprintf("%s: %s", prefix, line))
 			}
 		}
 	}(ctx, prefix)
 
 	return stream, func() {
+		w.WriteString("close\n")
```

Comment:
This seems like a workaround. Is there not a more direct approach?

Reply:
Perhaps.  There are two things going on here, getting the consumer not the producer to shut the pipe but also to ensure that the pipe is not just flushed but also empty, and the only way I know how to do that is to send some kind of tag through it.

Reply:
this seems okay as a temporary workaround IMO, we can follow-up with a more concrete fix

Reply:
I don't know what a more concrete fix would be, even using a semaphore to message from the producer to the consumer that it should exit there's still no way to flush a buffered pipe that I'm aware of.